pkg/api/README.md

```diff
@@ -0,0 +1,82 @@
+# OpenAPI specifications
+
+Since version 8.4, HTTP API details are [specified](https://editor.swagger.io/?url=https://raw.githubusercontent.com/grafana/grafana/main/public/api-merged.json) using OpenAPI v2. Starting from version 9.1, there is also a [OpenAPI v3 specification](https://editor.swagger.io/?url=https://raw.githubusercontent.com/grafana/grafana/main/public/openapi3.json) (generated by the v2 one using [this script](https://github.com/grafana/grafana/blob/main/scripts/openapi3/openapi3conv.go)).
+
+# OpenAPI annotations
+
+The OpenAPI v2 specification is generated automatically from the annotated Go code using [go-swagger](https://github.com/go-swagger/go-swagger). The code annotation rules can be found [here](https://goswagger.io/use/spec.html).
+
+Developers modifying the HTTP API endpoints need to make sure to add the necessary annotations so that their changes are reflected into the generated specifications.
+
+## Example of endpoint annotation
+
+The following defines a `PATCH` endpoint under the `/serviceaccounts/{serviceAccountId}` path with tag `service_accounts` (used for grouping together several routes) and operation id `updateServiceAccount` (used for uniquely identifying routes and associate parameters and response with them).
```

Comment:
I think the Google style for ID is "enter your ID" but "enter the id attribute", so ID would probably be preferred here.